src/components/QueryEditor/QueryEditor.test.tsx

```diff
@@ -100,177 +106,185 @@ async function renderWithQuery(query: Omit<XrayQuery, 'refId'>, rerender?: any)
+        await act(async () => {
+          render(
+            <QueryEditor
+              {...{
+                ...defaultProps,
+                datasource: {
+                  ...defaultProps.datasource,
+                  getAccountIdsForServiceMap: mockGetAccountIdsForServiceMap,
+                },
+                query: {
+                  refId: 'A',
+                  queryType: 'getServiceMap',
+                  accountIds: ['account1'],
+                } as any,
+              }}
+              onChange={() => {}}
+            />
+          );
+          expect(screen.getByText('', { selector: '.fa-spinner' })).toBeDefined();
+          await waitFor(() => expect(screen.getByText('account1')).toBeDefined());
+          expect(mockGetAccountIdsForServiceMap).toHaveBeenCalled();
+        });
+      });
 
-  it('does not fetch account ids if service map is not selected', async () => {
-    const mockGetAccountIdsForServiceMap = jest.fn(() => Promise.resolve(['account1', 'account2']));
-    await act(async () => {
-      render(
-        <QueryEditor
-          {...{
-            ...defaultProps,
-            datasource: {
-              ...defaultProps.datasource,
-              getAccountIdsForServiceMap: mockGetAccountIdsForServiceMap,
-            },
-            query: {
-              refId: 'A',
-              queryType: XrayQueryType.getTrace,
-              accountIds: [],
-            } as any,
-          }}
-          onChange={() => {}}
-        />
-      );
-      expect(screen.getByText('', { selector: '.fa-spinner' })).toBeDefined();
-      await waitForElementToBeRemoved(() => screen.getByText('', { selector: '.fa-spinner' }));
-      expect(mockGetAccountIdsForServiceMap).not.toHaveBeenCalled();
+      it('does not fetch account ids if service map is not selected', async () => {
+        const mockGetAccountIdsForServiceMap = jest.fn(() => Promise.resolve(['account1', 'account2']));
+        await act(async () => {
+          render(
+            <QueryEditor
+              {...{
+                ...defaultProps,
+                datasource: {
+                  ...defaultProps.datasource,
+                  getAccountIdsForServiceMap: mockGetAccountIdsForServiceMap,
+                },
+                query: {
+                  refId: 'A',
+                  queryType: XrayQueryType.getTrace,
+                  accountIds: [],
+                } as any,
+              }}
+              onChange={() => {}}
+            />
+          );
+          expect(screen.getByText('', { selector: '.fa-spinner' })).toBeDefined();
+          await waitForElementToBeRemoved(() => screen.getByText('', { selector: '.fa-spinner' }));
+          expect(mockGetAccountIdsForServiceMap).not.toHaveBeenCalled();
+        });
+      });
     });
-  });
+  }
+
+  run('QueryEditorForm with awsDatasourcesNewFormStyling disabled');
+  grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling = true;
```

Comment:
So I think as a best practice we want to make sure that regardless of the configuration of the machine running these tests, we are properly testing both scenarios. We don't want to make an assumption for example that grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling is false by default, because you might have it enabled on your machine. We also want to have some kind of clean up task so that there's no risk that this config could be set differently than expected in other tests down the line. Here's one way you could do that:

```
const originalValue = grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling;
afterEach(() => {
grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling = originalValue
})
describe("it should render properly when awsDatasourcesNewFormStyling is enabled, () => {
grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling = true
run()
})

describe("it should render properly when awsDatasourcesNewFormStyling is disabled, () => {
grafanaRuntime.config.featureToggles.awsDatasourcesNewFormStyling = false
run()
})
```

Reply:
Fixed it! Followed the same process like in Redshift and Athena if that sounds good :)